src/server/scripts/Zandalar/Underrot/boss_elder_leaxa.cpp

```diff
@@ -0,0 +1,385 @@
+    }
+};
+
+// 260889 - Creeping Rot
+class spell_creeping_rot_selector : public SpellScript
+{
+    PrepareSpellScript(spell_creeping_rot_selector);
+
+    static constexpr float SPAWN_DISTANCE = 5.0f;
+
+    // spell has SPELL_ATTR3_ONLY_ON_PLAYER, but its selecting the mirrors somehow
+    void FilterTargets(std::list<WorldObject*>& targets)
+    {
+        targets.remove_if([](WorldObject* target) -> bool
+        {
+            return !target->IsPlayer();
+        });
+    }
+
+    void HandleHit(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+
+        if (Unit* target = GetHitUnit())
+            caster->CastSpell(target, SPELL_CREEPING_ROT_SUMMON);
+
+        float angle = caster->GetOrientation();
+        float destX = caster->GetPositionX() + SPAWN_DISTANCE * std::cos(angle);
+        float destY = caster->GetPositionY() + SPAWN_DISTANCE * std::sin(angle);
+        float destZ = caster->GetPositionZ();
+        Position spawnVisualDest(destX, destY, destZ);
+        caster->CastSpell(spawnVisualDest, SPELL_CREEPING_ROT_SPAWN_VISUAL, true);
+    }
+
+    void Register() override
+    {
+        OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_creeping_rot_selector::FilterTargets, EFFECT_0, TARGET_UNIT_SRC_AREA_ENEMY);
+        OnEffectHitTarget += SpellEffectFn(spell_creeping_rot_selector::HandleHit, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 261496 - Creeping Rot (Aura)
+class spell_creeping_rot_aura : public AuraScript
+{
+    PrepareAuraScript(spell_creeping_rot_aura);
+
+    void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_EXPIRE)
+            return;
+
+        Creature* creature = GetTarget()->ToCreature();
+        if (!creature)
+            return;
+
+        creature->DespawnOrUnsummon();
+    }
+
+    void Register() override
+    {
+        AfterEffectRemove += AuraEffectRemoveFn(spell_creeping_rot_aura::OnRemove, EFFECT_0, SPELL_AURA_PERIODIC_TRIGGER_SPELL, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+static Position const MirrorSpawnPositions[] =
+{
+    { 879.79865f,   1222.6233f, 56.47815f  },
+    { 856.184f,     1232.5435f, 56.52007f  },
+    { 859.34204f,   1238.0243f, 56.520065f },
+    { 857.8333f,    1220.6545f, 56.468567f },
+    { 876.17365f,   1240.2848f, 56.409615f },
+    { 864.25867f,   1240.5903f, 56.520065f },
+    { 863.1042f,    1218.0591f, 56.468765f },
+};
+
+// 264603 - Blood Mirror
+class spell_blood_mirror_selector : public SpellScript
+{
+    PrepareSpellScript(spell_blood_mirror_selector);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_BLOOD_MIRROR_MISSILE });
+    }
+
+    void HandleHit(SpellEffIndex /*effIndex*/)
+    {
+        Position spawnPos = Trinity::Containers::SelectRandomContainerElement(MirrorSpawnPositions);
```

Comment:
If I recall correctly, the boss wouldn't ever choose the closest spawn point from its own position of all possible choices. Maybe it's best to check it out on videos or wait for next season; in any case, you'd need to obtain the members, compare based on distance, create a new container and exclude the closest to the boss and then obtain a random member from it.

Reply:
just confirmed its random; one mirror literally spawned next to me